ffpuppet/core.py

```diff
@@ -650,22 +652,52 @@ def create_profile(extension=None, prefs_js=None, template=None):
                 with open(times_json, "w") as times_fp:
                     times_fp.write('{"created":%d}' % (int(time.time()) * 1000))
 
-        # XXX: fuzzpriv extension support
-        # should be removed when bug 1322400 is resolved if it is no longer used
-        if extension is not None:
-            os.mkdir(os.path.join(profile, "extensions"))
-            if os.path.isfile(extension) and extension.endswith(".xpi"):
-                shutil.copyfile(
-                    extension,
-                    os.path.join(profile, "extensions", os.path.basename(extension)))
-            elif os.path.isdir(extension):
-                shutil.copytree(
-                    os.path.abspath(extension),
-                    os.path.join(profile, "extensions", "domfuzz@squarefree.com"))
+        # extension support
+        try:
+            if extension is None:
+                extensions = []
+            elif isinstance(extension, (list, tuple)):
+                extensions = extension
             else:
-                shutil.rmtree(profile, True) # clean up on failure
-                raise RuntimeError("Unknown extension: %r" % extension)
-
+                extensions = [extension]
+            if extensions:
```

Comment:
Should also check if os.path.join(profile, "extensions") exists

Reply:
Yup, thanks. Forgot about the template case.